tracer/samples/Samples.Transport.UnixDomainSocket/Program.cs

```diff
@@ -0,0 +1,33 @@
+using Datadog.Trace.TestHelpers;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Hosting;
+
+namespace Samples.Transport.UnixDomainSocket
+{
+    public class Program
+    {
+        public static MockTracerAgent MockAgent;
+
+        public static void Main(string[] args)
+        {
+            SetMockAgent();
+            CreateHostBuilder(args).Build().Run();
+        }
+
+        public static IHostBuilder CreateHostBuilder(string[] args) =>
+            Host.CreateDefaultBuilder(args)
+                .ConfigureWebHostDefaults(webBuilder =>
+                {
+                    webBuilder.UseStartup<Startup>();
+                });
+
+        private static void SetMockAgent()
+        {
+            MockAgent = new MockTracerAgent(
+                new UnixDomainSocketConfig(
+                    "/var/run/datadog/mockapm.socket",
+                    "/var/run/datadog/mockdsd.socket")
+                );
```

Comment:
This could live somewhere other than samples, but I used this sample as the primary development/debugging tool.
I focused on the F5 experience and wanted to be able to debug the `MockTracerAgent` and tracer code from what would be the happy path for UDS (*nix)... For that locally, I needed a container.

I started by using only the real agent, to be sure the client code was good. Then moved to develop the mock agent in here to be sure it was good... And then expanded to the tests.

It shortened my feedback cycles... as I didn't really want to wait on CI or develop on a *nix machine.

Reply:
Yeah, that's fine by me. I guess "Samples" is where we point users to in the docs, and this seems like it would be more harmful for them to see than useful 😄 That said, putting it in samples keeps it of the sln and CIs way, so 🤷‍♂️

Reply:
I'm happy to make a new folder that is uh... "Debugging"? Something along those lines, I am not sure what to name it.

Reply:
This sounds like the original "tools" folder 😆

Reply:
I mean... I liked it...

Reply:
I'm also a fan of the "tools" folder, and this seems appropriate to me

Reply:
The thing with "tools" is that, as far as I can tell, noone ever used them after the single use case they were built for (with the exception of the ones we moved to Nuke)... I suspect this might be in the same boat 😬